File: algorithms/python3/1_TwoSum_Easy/main.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i):
                if (nums[i] + nums[j] == target):
                    return [j, i]

n1 = [2,7,11,15]
t1 = 9
n2 = [3,2,4]
t2 = 6
n3 = [3,3]
t3 = 6

s = Solution()
print(s.twoSum(n1, t1))
print(s.twoSum(n2, t2))
print(s.twoSum(n3, t3))

File: algorithms/cpp/1_TwoSum_Easy/main.cpp
#include <iostream>
#include <vector>

using namespace std;


vector<int> twoSum(vector<int>& nums, int target) {

    for (int i = 0; i < nums.size(); i++) {
        for (int j = 0; j < i; j++) {
            if (nums[i] + nums[j] == target) {
                return vector<int> {j, i};
            }
        }
    }


    return nums;
}


int main() {
    
    vector<int> v1 {2,7,11,15};
    int t1 = 9;
    vector<int> v2 {3,2,4};
    int t2 = 6;
    vector<int> v3 {3,3};
    int t3 = 6;

    cout << "[" << twoSum(v1,t1)[0] << ", " << twoSum(v1, t1)[1] << "]\n";

    cout << "[" << twoSum(v2,t2)[0] << ", " << twoSum(v2, t2)[1] << "]\n";

    cout << "[" << twoSum(v3,t3)[0] << ", " << twoSum(v3, t3)[1] << "]\n";



    return 0;
}